app/static/assets/js/view-grades.js

```diff
@@ -132,20 +280,29 @@ $(function () {
                 }
             });
         } else {
+            data = response.map(item => ({'id': item, 'text': item}));
             dropdown.empty().prepend('<option></option>').select2({
-                data: response.map(item => ({'id': item, 'text': item})),
+                data: data,
             });
         }
-        // If the response contains only one element or the section is length 2 with OVERALL, automatically select it
-        if (id === "section" && response.length === 2 && response.includes("OVERALL")) {
-            // Find the entry that is not OVERALL
+
+        // If the previous selection was found, trigger the selection automatically
+        if (data.find((e) => e.id === previousSelection)) {
             dropdown.select2("trigger", "select", {
-                data: {id: response.filter(x => x !== "OVERALL")[0]}
+                data: {
+                    id: previousSelection,
+                }
             });
-            $('#vg-dropdown-form').submit();
         } else {
-            // Open the dropdown
-            dropdown.select2('open');
+            // If the response contains only one element or the section is length 2 with OVERALL, automatically select it
+            if (id === "section" && response.length === 2 && response.includes("OVERALL")) {
+                // Find the entry that is not OVERALL
+                dropdown.select2("trigger", "select", {
+                    data: {id: response.filter(x => x !== "OVERALL")[0]}
+                });
+            } else {
+                dropdown.select2("open");
+            }
```

Comment:
I just realized that the dropdown handling here should be refactored out into the `StateHandler`. This method should probably only update the dropdown and return the formed data so the handler can diff and make a decision there.

I'll update this PR tomorrow.